### PR TITLE
Implement food category scales based scoring.

### PR DESCRIPTION
**Pull request by n1k0**
[User story](https://www.notion.so/RESULTATS-Introduire-des-chelles-intra-cat-gorie-pour-la-note-A-B-C-D-E-c3b7e176d44d4be985612ccf2dce3266)

### CODE REVIEW

File: src/Data/Impact.elm [hunk 1]
```diff
@@ -418,10 +421,44 @@ getAggregatedScoreOutOf100 { trigram } impactsPerKg =
...
+getAggregatedCategoryScoreOutOf100 : Definition -> String -> Impacts -> Int
+getAggregatedCategoryScoreOutOf100 { trigram } foodCategory impactsPerKg =
+    case Dict.get foodCategory foodCategories of
+        Just { bounds } ->
+            let
+                ln =
+                    logBase e
+
+                ( min, max ) =
+                    bounds
```
Comment (magopian) [inline src/Data/Impact.elm:446]: Should we use the same names as in the docs, so `( impact100, impact0) = bounds` in this case? I had a hard time wrapping my head around the formula in the docs, and it didn't seem to fit with the formula I read below (took me some thinking to understand that `min` was in fact `impact100`)

File: src/Data/Impact.elm [hunk 2]
```diff
@@ -418,10 +421,44 @@ getAggregatedScoreOutOf100 { trigram } impactsPerKg =
...
+                |> Unit.impactToFloat
+                |> (\value ->
+                        -- See docs at https://fabrique-numerique.gitbook.io/ecobalyse/alimentaire/impacts-consideres/score-100#projet-declinaisons-du-score-100
+                        (ln (toFloat max) - ln value) / ln (toFloat max / toFloat min) * 20 * 5
+                   )
+                |> floor
+                |> clamp 0 100
+
+        Nothing ->
+            -99
```
Comment (magopian) [inline src/Data/Impact.elm:459]: This makes me kinda uncomfortable... I understand it's an "impossible value"... would we want to use `0` which is still in the range of "acceptable scores", or instead return a maybe or a result?

Not sure about this, just a feeling, a "smell"

**Review by magopian — APPROVED** [review]
Excellent work, thanks! A minor nit and an open question, you get to decide what to do with them ;)

File: src/Data/Impact.elm (hunk 1 above)
Comment (n1k0) [reply src/Data/Impact.elm:464]: https://github.com/MTES-MCT/ecobalyse/pull/210/commits/b624b0c543a31e6bf8b9870387fb2b22d013ae90

File: src/Data/Impact.elm (hunk 2 above)
Comment (n1k0) [reply src/Data/Impact.elm]: image

Fixed in https://github.com/MTES-MCT/ecobalyse/pull/210/commits/d975ddbab2add9dc2d117ac03c3ecccea0db959e

**Pull request merged**